### PR TITLE
Temporarily Apply Configuration for Current Run from Command Line

### PR DESCRIPTION
**Pull request by cockroacher**
### Feature/improvement

Issue: #488

I propose a feature that allows users to temporarily apply a configuration for the current run from the command line. This would provide greater flexibility and control over individual runs without having to modify any configuration files.

The proposed feature would allow users to override the default settings for a single run by providing them as command-line arguments. This would be particularly useful for testing different configurations without having to modify any configuration files.

Moreover, this feature could potentially minimize the need for users to create their own configuration files. Instead, they could quickly and easily adjust the settings for each run as needed.

### Examples
Here are a couple of examples where this feature could be useful:

1. **Detailed Report**: The `USE_DETAILED_REPORT` configuration determines whether tests use a detailed report for review and rating when available. By default, this is set to `False`. However, a user might want to enable this for a specific run to get more detailed insights. With the proposed feature, they could do this directly from the command line without having to modify the configuration file.

    ```bash
   python default.py -r -i sites.json --setting details=true
    ```

2. **Show Improvements Only**: The `REVIEW_SHOW_IMPROVEMENTS_ONLY` configuration determines whether only reviews that can be improved are shown. By default, this is set to `False`. However, a user might want to enable this for a specific run to focus on areas of improvement. With the proposed feature, they could do this directly from the command line without having to modify the configuration file.

    ```bash
   python default.py -r -i sites.json --setting improve-only=true
    ```

Looking forward to your thoughts on this.

### CODE REVIEW

File: default.py [hunk 1]
```diff
@@ -251,6 +344,60 @@ def set_input_take(self, arg):
...
+
+        config_value = None
+        if value in ('true', 'True', 'yes', 'Y', 'y'):
+            config_value = True
+        elif value in ('false', 'False', 'no', 'N', 'n'):
+            config_value = False
+        elif value in ('none', 'None', 'NONE'):
+            config_value = None
+        else:
+            print('no valid configuration value for:', config_name)
```
Comment (7h3Rabbit) [inline default.py:396]: @cockroacher If I read this correctly this would not allow us to set string settings, for example:
`python default.py -i sites.json --setting browser=firefox`

Comment (7h3Rabbit) [reply default.py:396]: @cockroacher 
`python default.py -i sites.json --setting tests.sitespeed.iterations=5`
This doesn't work either I guess?
(Havn't tested)

Comment (7h3Rabbit) [reply default.py:396]: @cockroacher
could we make the default `config_value = True` (and allow for empty value)?
Then we could write stuff like:
`python default.py -i sites.json --setting improve-only`
instead of:
`python default.py -i sites.json --setting improve-only=true`

File: default.py [hunk 2]
```diff
@@ -426,6 +573,7 @@ def handle_option(self, opt, arg):
             ("--input-take"): self.set_input_take,
             ("-o", "--output"): self.set_output_filename,
             ("-r", "--review", "--report"): self.enable_reviews,
+            ("--setting"): self.set_setting
```
Comment (7h3Rabbit) [inline default.py:576]: @cockroacher 
lets use `-s` as shorthand.

File: default.py [hunk 3]
```diff
@@ -505,9 +656,7 @@ def main(argv):
             # Cleanup exipred cache
         clean_cache_files()
     else:
-        print(options.language('TEXT_COMMAND_USAGE'))
-
-
+        options.show_help(_)
```
Comment (7h3Rabbit) [inline default.py:659]: @cockroacher 
Why this change?

File: helpers/tls_helper.py [hunk 4]
```diff
@@ -28,20 +23,24 @@ def rate_transfer_layers(result_dict, global_translation, local_translation, dom
     Returns:
         Rating: A Rating object that represents the rating of the domain's transport layers.
     """
-    rating = Rating(global_translation, REVIEW_SHOW_IMPROVEMENTS_ONLY)
+    rating = Rating(global_translation, get_config_or_default('review_show_improvements_only'))
```
Comment (7h3Rabbit) [inline helpers/tls_helper.py:26]: @cockroacher 
Why don't we use `setting_helper.get_config(name)` directly?

File: default.py (hunk 1 above)
Comment (cockroacher) [reply default.py]: Fixed, thanks :)

File: default.py (hunk 3 above)
Comment (cockroacher) [reply default.py]: unintended change, reverted

**Comment by 7h3Rabbit** [comment]
https://github.com/Webperf-se/webperf_core/blob/main/docs/settings-json.md
